Tests intermittently fail: discrepancies between Python and C++ versions

The failure message is:
```
_________________ TestBloomFilterComparison.test_small_input_a _________________
self = <tests.test_similarity.TestBloomFilterComparison testMethod=test_small_input_a>
    def test_small_input_a(self):
        py_similarity = entitymatch.calculate_filter_similarity(
            self.filters1[:10], self.filters2, self.default_k, self.default_threshold, use_python=True)
        c_similarity = entitymatch.calculate_filter_similarity(
            self.filters1[:10], self.filters2, self.default_k, self.default_threshold, use_python=False)
>       self.assert_similarity_matrices_equal(py_similarity, c_similarity)
tests/test_similarity.py:93: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_similarity.py:40: in assert_similarity_matrices_equal
    self.assertIn(m, N)
E   AssertionError: (4, 0.6204311152764761, 32) not found in [(0, 0.6625222024866785, 58), (0, 0.6623012160898035, 121), (0, 0.6598639455782312, 67), (0, 0.6573556797020484, 45), (0, 0.6557692307692308, 79), (0, 0.6515837104072398, 38), (0, 0.6469465648854962, 146), (0, 0.642791551882461, 174), (0, 0.6415441176470589, 118), (0, 0.6411815812337098, 108), (1, 0.6359832635983264, 124), (1, 0.6077003121748179, 131), (1, 0.6065573770491803, 82), (1, 0.5901309164149043, 38), (1, 0.5859213250517599, 57), (1, 0.5857142857142857, 141), (1, 0.5821989528795811, 20), (1, 0.5788923719958202, 98), (1, 0.5777351247600768, 51), (1, 0.5733186328555678, 80), (2, 0.6559849198868991, 141), (2, 0.6553267681289168, 113), (2, 0.6489945155393053, 150), (2, 0.6465028355387523, 174), (2, 0.6458141674333027, 53), (2, 0.6443202979515829, 38), (2, 0.6356435643564357, 182), (2, 0.63003663003663, 12), (2, 0.6278625954198473, 147), (2, 0.6258124419684308, 73), (3, 0.6483931947069943, 62), (3, 0.6425166825548141, 181), (3, 0.6388888888888888, 103), (3, 0.6383799421407907, 57), (3, 0.6350877192982456, 167), (3, 0.6328413284132841, 150), (3, 0.6299810246679317, 156), (3, 0.6290471785383904, 162), (3, 0.6238894373149062, 7), (3, 0.623574144486692, 56), (4, 0.6557377049180327, 14), (4, 0.6514598540145985, 165), (4, 0.6353149955634427, 113), (4, 0.6319772942289499, 40), (4, 0.6272727272727273, 194), (4, 0.6267190569744597, 41), (4, 0.6265938069216758, 135), (4, 0.624380574826561, 43), (4, 0.6210221793635486, 179), (4, 0.6204311152764761, 118), (5, 0.6576663452266153, 159), (5, 0.6387755102040816, 85), (5, 0.6307222787385555, 42), (5, 0.6305506216696269, 67), (5, 0.6301633045148896, 76), (5, 0.6266266266266266, 190), (5, 0.62426614481409, 185), (5, 0.6229508196721312, 75), (5, 0.6210221793635486, 166), (5, 0.6183953033268101, 122), (6, 0.6543095458758109, 58), (6, 0.650132860938884, 67), (6, 0.6387035271687321, 153), (6, 0.6359269932756965, 82), (6, 0.631578947368421, 123), (6, 0.6282527881040892, 12), (6, 0.6274509803921569, 157), (6, 0.6266924564796905, 15), (6, 0.624618514750763, 85), (6, 0.6194503171247357, 96), (7, 0.6476377952755905, 89), (7, 0.6458715596330276, 72), (7, 0.6454935622317597, 67), (7, 0.6453433678269049, 45), (7, 0.6438848920863309, 12), (7, 0.6432748538011696, 114), (7, 0.6389891696750902, 14), (7, 0.6368613138686131, 87), (7, 0.6345226615236258, 146), (7, 0.6319702602230484, 166), (8, 0.6720430107526881, 63), (8, 0.6517690875232774, 157), (8, 0.6481481481481481, 131), (8, 0.6429840142095915, 2), (8, 0.6412078152753108, 88), (8, 0.639269406392694, 82), (8, 0.6312900274473925, 139), (8, 0.6306976744186047, 52), (8, 0.6305506216696269, 14), (8, 0.6303030303030303, 75), (9, 0.6423982869379015, 164), (9, 0.6328600405679513, 15), (9, 0.6215943491422805, 106), (9, 0.6086956521739131, 67), (9, 0.6015779092702169, 97), (9, 0.5985267034990792, 167), (9, 0.5981873111782477, 32), (9, 0.5946524064171123, 85), (9, 0.593654042988741, 185), (9, 0.5919661733615222, 100)]
```

This tests checks if the results returned by the Python version and the C++ version are the same. Clearly sometimes they aren’t.

--- a/_cffi_build/dice_one_against_many.cpp
+++ b/_cffi_build/dice_one_against_many.cpp
@@ -262,7 +262,7 @@ public:
 
 struct score_cmp {
     bool operator()(const Node& a, const Node& b) const {
-        return a.score >= b.score;
+        return a.score > b.score || (a.score == b.score && a.index < b.index);
     }
 };
 

--- a/anonlink/entitymatch.py
+++ b/anonlink/entitymatch.py
@@ -29,7 +29,7 @@ def python_filter_similarity(filters1, filters2, k, threshold):
 
         coeffs = filter(lambda c: c[1] >= threshold,
                         enumerate(map(dicecoeff, filters2)))
-        top_k = sorted(coeffs, key=itemgetter(1), reverse=True)[:k]
+        top_k = sorted(coeffs, key=lambda x: -x[1])[:k]
         result.extend([(i, coeff, j) for j, coeff in top_k])
     return result
 

